sigstore/_utils.py

```diff
@@ -139,6 +141,13 @@ def key_id(key: PublicKey) -> KeyID:
 
     return KeyID(hashlib.sha256(public_bytes).digest())
 
+def get_digest(input_: IO[bytes] | sigstore_hashes.Hashed) -> sigstore_hashes.Hashed:
+    if isinstance(input_, sigstore_hashes.Hashed):
+        return input_
+
+    # NOTE: Not able to check for the type `TypeError: Subscripted generics cannot be used with class and instance checks`
+    # when calling isinstance(_input, IO[bytes])
```

Comment:
Extreme nitpick, but IMO we can drop this `NOTE`: the `sha256_streaming` call's parameter hint will keep this checked for us.